app/models/order.rb

```diff
@@ -11,7 +11,7 @@ class Order < ApplicationRecord
   has_many :deals, through: :line_items
   
   scope :placed_orders, ->{ includes(:address, :payments).where(orders: { status: 'Placed' }).where(payments: { status: 'Successful' }) }
-  scope :deal_exists, ->(deal_id){ includes(:deals).where(deals: { id: deal_id }) }
+  scope :deal_exists, ->(deal_id){ joins(:deals).where(deals: { id: deal_id }) }
```

Comment:
what's the difference between includes and joins here?

Reply:
Here we are using deals to only filter orders having the received deal but we actually don't need the data from Deals table, so instead of includes ,we can use joins as we don't need to eagerly load deals Data.